dist/gcc-compatible/evercrypt_targetconfig.h

```diff
@@ -6,27 +6,22 @@
  * - https://sourceforge.net/p/predef/wiki/Architectures/
  */
 
-/* TODO: this should be overhauled to use the outcome of the configure script,
- * found in config.h, exclusively, rather than doing feature tests via
- * preprocessor macros. In particular, the only relevant information for
- * EverCrypt should be: SUPPORTS_128 and SUPPORTS_256. */
-
-#if defined(__x86_64__) || defined(_M_X64)
-#define EVERCRYPT_TARGETCONFIG_X64 1
-#elif defined(__i386__) || defined(_M_IX86)
-#define EVERCRYPT_TARGETCONFIG_X86 1
-#elif defined(__aarch64__) || defined(_M_ARM64)
-#define EVERCRYPT_TARGETCONFIG_AARCH64 1
-#elif defined(__arm__) || defined(_M_ARM)
-#define EVERCRYPT_TARGETCONFIG_AARCH32 1
-#endif
+enum {
+  // TODO: the macros defined in EverCrypt.TargetConfig.h don't extract as
+  // uppercase - fix that
+  target_architecture_name_x86,
+  target_architecture_name_x64,
+  target_architecture_name_arm7,
+  target_architecture_name_arm8,
+  target_architecture_name_systemz,
+  target_architecture_name_powerpc64,
+  target_architecture_name_unknown
+};
```

Comment:
If you can extend Kremlin to extract uppercase if something like `[@CMacro]` is above the definition, that would be great.